src/administration/install_docker.md

```diff
@@ -1,78 +1,118 @@
+
+Ensure that a modern version of `docker` and `docker-compose` (>=`1.24.0`) are available in the environment. On Ubuntu, this is accomplished by executing `apt install docker-compose docker.io`; other distributions may have different package names.
+
+### Reverse Proxy / Webserver
+
+Lemmy requires the use of a Reverse Proxy to properly direct requests to either the Lemmy server or Lemmy UI. This is because Lemmy is composed of two separate components, the server and the UI, which are served from different containers (or services in Docker Swarm mode) and different ports. The reverse proxy is responsible for directing requests to the correct component based on the URL, and also provides SSL termination for the proxied service.
+
+Some common choices include webservers such as [NGINX](https://www.nginx.com/), or a dedicated proxy such as [Traefik](https://traefik.io/). Configuration is _not_ limited to these two options, however these are the most common and example configurations are provided for them.
+
+The Reverse proxy may be part of the same Docker Compose file as the Lemmy server and UI, or it may be a separate service. In a production configuration it is recommended that the reverse proxy be a separate service, as this allows for greater flexibility in scaling and configuration.
+
+## Deployment
+
+Create a location to store the Lemmy docker environment; set any required permissions.
 
 ```bash
 # Create a folder for the lemmy files. the location doesnt matter, you can put this anywhere you want
-mkdir lemmy
-cd lemmy
+mkdir -p /data/lemmy/{volumes,config}
+mkdir -p /data/lemmy/volumes/{pictrs,postgres}
+chown 991:991 /data/lemmy/volumes/pictrs
+cd /data/lemmy
 ```
 
-## Download default config files
+### Download an appropriate Docker template
+
+There are many ways that the Lemmy docker environment can be configured. The following sections describe some of the more common configurations.
 
-These files contain `{{ }}` braces for variables, such as passwords and your domain.
+#### NGINX
 
-Edit them before starting up lemmy for the first time.
+Appropriate templates for nginx are available in the [lemmy-ansible](https://github.com/LemmyNet/lemmy-ansible) repository. While these are designed to be deployed via Ansible automation, the templates can be trivially hand-edited.
 
-The images will likely be: [dessalines/lemmy:VERSION](https://hub.docker.com/r/dessalines/lemmy) and [dessalines/lemmy-ui:VERSION](https://hub.docker.com/r/dessalines/lemmy-ui)
 
 ```bash
-wget https://raw.githubusercontent.com/LemmyNet/lemmy-ansible/main/templates/docker-compose.yml
-wget https://raw.githubusercontent.com/LemmyNet/lemmy-ansible/main/examples/config.hjson -O lemmy.hjson
-wget https://raw.githubusercontent.com/LemmyNet/lemmy-ansible/main/templates/nginx_internal.conf
+curl -O https://raw.githubusercontent.com/LemmyNet/lemmy-ansible/main/templates/docker-compose.yml
+pushd config
```

Comment:
What's going on here? All these should be at the same level.

Reply:
It really doesn't matter where the files get mounted in from; I thought it more explicit for documentation purposes to use a `./config` directory to bind in the configurations. We're already instructing users to pull and amend the `docker-compose.yml` template for Ansible for their non-Ansible deployments anyway...